lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -543,44 +543,51 @@ internal class EvaluatingCompiler(
         }
 
     private fun compilePlus(expr: PartiqlAst.Expr.Plus, metas: MetaContainer): ThunkEnv {
+        if (expr.operands.size < 2) {
+            error("Internal Error: PartiqlAst.Expr.Plus must have at lest 2 arguments")
+        }
+
         val argThunks = compileAstExprs(expr.operands)
 
-        val computeThunk = when (argThunks.size) {
-            //Unary +
-            1 -> {
-                val firstThunk = argThunks[0]
-                thunkFactory.thunkEnvOperands(metas, firstThunk) { _, value ->
-                    //Invoking .numberValue() here makes this essentially just a type check
-                    value.numberValue()
-                    //Original value is returned unmodified.
-                    value
-                }
-            }
-            //N-ary +
-            else -> thunkFactory.thunkFold(metas, argThunks) { lValue, rValue ->
-                (lValue.numberValue() + rValue.numberValue()).exprValue()
-            }
+        val computeThunk = thunkFactory.thunkFold(metas, argThunks) { lValue, rValue ->
+            (lValue.numberValue() + rValue.numberValue()).exprValue()
         }
 
         return resolveIntConstraint(computeThunk, metas)
     }
 
     private fun compileMinus(expr: PartiqlAst.Expr.Minus, metas: MetaContainer): ThunkEnv {
+        if (expr.operands.size < 2) {
+            error("Internal Error: PartiqlAst.Expr.Plus must have at lest 2 arguments")
```

Comment:
A couple typos here. `Plus` -> `Minus` and "lest" -> "least"

```suggestion
            error("Internal Error: PartiqlAst.Expr.Minus must have at least 2 arguments")
```